[Fix] 메서드 제한자 오류

## 🐞 Bug Description  
SimpleDiagnosisDTO의 of 메서드가 default로 생성되어 오류 발생

## ✅ Expected Behavior  
Service에서 of 메서드를 사용

## 🔁 Reproduce  
현재 어떤 문제가 발생했는지 설명해주세요
1. DiagnosisService에서 오류 발생

## 🖼️ Screenshots  
가능하다면 문제를 보여주는 스크린샷을 첨부해주세요.

<img width="767" height="195" alt="Image" src="https://github.com/user-attachments/assets/c2bd0ce7-2e88-4c16-b2a5-695c5000a261" />

## 🧩 Additional Context  
C:\KB\Back-End\src\main\java\cheongsan\domain\policy\service\DiagnosisServiceImpl.java:106: error: of(SimpleDiagnosis) is not public in SimpleDiagnosisDTO; cannot be accessed from outside package
        SimpleDiagnosisDTO simpleDiagnosisDTO = SimpleDiagnosisDTO.of(simpleDiagnosis);
                                                                  ^

--- a/src/main/java/cheongsan/domain/policy/dto/SimpleDiagnosisDTO.java
+++ b/src/main/java/cheongsan/domain/policy/dto/SimpleDiagnosisDTO.java
@@ -18,7 +18,7 @@ public class SimpleDiagnosisDTO {
     private String simpleDescription;
 
     // Diagnosis Entity → DTO 변환 static 메서드 (Builder 활용)
-    static SimpleDiagnosisDTO of(SimpleDiagnosis simpleDiagnosis) {
+    public static SimpleDiagnosisDTO of(SimpleDiagnosis simpleDiagnosis) {
         return SimpleDiagnosisDTO.builder()
                 .id(simpleDiagnosis.getId())
                 .programName(simpleDiagnosis.getProgramName())

